go/vt/vtgate/resolver.go

```diff
@@ -46,17 +46,19 @@ var (
 // Resolver is the layer to resolve KeyspaceIds and KeyRanges
 // to shards. It will try to re-resolve shards if ScatterConn
 // returns retryable error, which may imply horizontal or vertical
-// resharding happened.
+// resharding happened. It is implemented using a srvtopo.Resolver.
```

Comment:
I'm wondering if this resolver can be obsoleted. Or we could consider renaming it.

Reply:
Yes I agree, we may be able to get rid of it.